tests/protocol/test_imds.py

```diff
@@ -234,45 +239,55 @@ def _setup_image_origin_assert(publisher, offer, sku, version):
 
         return compute_info.image_origin
 
-    def test_response_validation(self):
+    @patch("azurelinuxagent.common.protocol.util.ProtocolUtil")
+    def test_response_validation(self, ProtocolUtil):
+        ProtocolUtil().get_wireserver_endpoint.return_value = "foo.bar"
+
         # invalid json or empty response
         self._assert_validation(http_status_code=200,
                                 http_response='',
                                 expected_valid=False,
-                                expected_response='JSON parsing failed')
+                                expected_response='JSON parsing failed',
+                                expected_fallback=False)
 
         self._assert_validation(http_status_code=200,
                                 http_response=None,
                                 expected_valid=False,
-                                expected_response='JSON parsing failed')
+                                expected_response='JSON parsing failed',
+                                expected_fallback=False)
 
         self._assert_validation(http_status_code=200,
                                 http_response='{ bad json ',
                                 expected_valid=False,
-                                expected_response='JSON parsing failed')
+                                expected_response='JSON parsing failed',
+                                expected_fallback=False)
 
         # 500 response
         self._assert_validation(http_status_code=500,
                                 http_response='error response',
                                 expected_valid=False,
-                                expected_response='[HTTP Failed] [500: reason] error response')
+                                expected_response='[HTTP Failed] [500: reason] error response',
+                                expected_fallback=True)
```

Comment:
TODO: Only fallback on errors listed in `azurelinuxagent.common.utils.restutil.RETRY_EXCEPTIONS`